src/Corrade/Utility/DebugStl.h

```diff
@@ -103,9 +103,32 @@ template<class ...Args> Debug& operator<<(Debug& debug, const std::tuple<Args...
 
 namespace Implementation {
 
+template<typename T>
+typename std::add_lvalue_reference<T>::type declare_lvalue_reference() noexcept;
```

Comment:
This is basically "like `std::declval<T>()`, but returning a T& instead of T&&", right? :) I think this might be useful in other contexts as well, will put it into `TypeTraits.h` as a public API.

Reply:
yes! `std::delcval` declares an rvalue refrence, so it picked up the `operator<<` for `Debug&&` instead of the ones wanted